src/main/java/net/fosforito/partido/util/Pair.java

```diff
@@ -0,0 +1,19 @@
+package net.fosforito.partido.util;
+
+public final class Pair<A, B> {
```

Comment:
I think that an object containing a validation message and an object is definitely nice to have, specially if there come more validation cases later. But the usage of this class is not really self explaining.

I would suggest to rename `Pair` with something more specific like `Result` or similar because it contains a result message and an object. `getFirst()` would become `getResultMessage()` and `getSecond()` `getObject()` or similar.

With these changes we avoid creating an "utility" class that has no `Single Responsibility` according to Clean Code.

Reply:
The goal isn't to have Pair as a result primarily. It's a utility class and can be used for other purposes as well.
Primarily calling it a Result would imply it's only a result, whilst it could also be a transfered Pair of 2 Strings.

Reply:
The Pair class is more flexible than a Result but the problem is here that because of the abstraction the code becomes harder to read.

```
Pair<String, User> validationOfCreatedUser = userService.createUser(userDTO);
    if (validationOfCreatedUser.getFirst() == null) {
      return [...] validationOfCreatedUser.getSecond() [...];
    }
```

I personally didn't understand what exactly was happening here without looking deeper into the UserService and than into the Pair class. Even without looking into the methods and types used by a single function, others should understand what's going on.

I see two options how we can improve this:
- create a more concrete class implementing the Pair (interface) used to transfer Results containing a message and an object or
- create and use a different class directly for communication between *Api classes and *Service classes.

Anyway I would suggest to roll back this change and undo adding the UserService class and I'll open a new issue to discuss options. This makes sure that all *Api classes will have the same structure as the changed one and that Pull Requests do not fix or change everything everything around the main issue at once.